### PR TITLE
mliap interface works with hipnn-coulomb architecture

### PR DESCRIPTION
**Pull request by sakibmatin**
The `mliap` interface uses per-atom energies. 
The `CombineEnergy` node (PR contribution) now allows the addition of per-atom energies from two nodes, e.g.  `HEnergy` with  `CoulombEnergy` or `ScreenedCoulombEnergy`

Updated sketch of  how to utilize this code when building hipnn models :

```python
# Screened Coulomb Energy using predicted charges
  coulomb_energy = physics.ScreenedCoulombEnergyNode(
      "coulomb_Energy",
      (atom_charges,
          pairfinder.pair_dist, pairfinder.pair_first, pairfinder.pair_second,
          padidxer.mol_index, padidxer.n_molecules),
      energy_conversion=energy_conv, 
      screening=combined_screening, 
      cutoff_distance=coulomb_r_max
  )

# Short range Energy contribution
Henergy = targets.HEnergyNode("HEnergy", network_energy) 

# Combined Energy
sys_energy = physics.CombineEnergyNode("sysEnergy", 
    (Henergy, coulomb_energy),  # Padding Indexer is automatically extracted from connected graph. 
    db_name="energy"
)
```
Then the `sys_energy` is energy node when constructing calculators for ASE or Lammps.

### CODE REVIEW

File: hippynn/layers/physics.py [hunk 1]
```diff
@@ -106,37 +106,36 @@ def __init__(self, energy_conversion_factor, screening, radius=None):
...
+        voltage_pairs = self.energy_conversion_factor * (charges[pair_second] / pair_dist.unsqueeze(1)) 
+        voltage_pairs = voltage_pairs * screening 
+        n_atoms, _ = charges.shape
+        voltage_atom = torch.zeros((n_atoms, 1), device=charges.device, dtype=charges.dtype)
+        voltage_atom.index_add_(0, pair_first, voltage_pairs) 
+        coulomb_atom = voltage_atom * charges
+        coulomb_molecules = 0.5 * self.summer(coulomb_atom, mol_index, n_molecules)
 
-        return coulomb_molecule
+        return coulomb_molecules, coulomb_atom
```
Comment (lubbersnick) [inline hippynn/layers/physics.py]: Isn't this double-counting on the atom level? In HEnergy, atom energies are run through the system summation (`self.summer`) to produce the total energy. In this case there is a factor of 0.5 between them.

**Comment by lubbersnick** [comment]
Please do not worry about the job failures for the moment, they are a reflection of the fact that I do not yet know how to put the settings properly.

File: hippynn/layers/physics.py (hunk 1 above)
Comment (sakibmatin) [reply hippynn/layers/physics.py]: In the new commits, this issue is resolved. 
Coulomb energy is defined for a pair of atoms. Here, we adopt the  convention that the Coulomb energy for a pair of atoms is evenly partitioned to both atoms as the 'per-atom energies'. Therefore, the atom energies sum to the molecular energy; similar to the `HEnergy`.

**Comment by sakibmatin** [comment]
I believe that this PR is ready to be merged.

**Comment by lubbersnick** [comment]
Made some parent expansion changes to the node building. If this doesn't work right let me know. It should now work to make the coulomb node with only the charge as parents, e.g.
```
coul = CoulombEnergyNode("coulomb energy",hcharge,energy_conversion=1)
```
Before there was a bug so this form of construction couldn't work.

**Comment by sakibmatin** [comment]
Your commits works with the `physics.CombineEnergyNode`. 
 I've also tested that the screened coulomb energy node can be constructed from just the charge node. 
 ``` python
 screen_coul = physics.ScreenedCoulombEnergy(
     "screened coulomb energy",
     (hcharge), 
     energy_conversion=1,
     screening=screening,
     cutoff_distance=coulomb_cutoff
 )
 ```
I believe, that this PR can be merged.

**Pull request merged**